### ISSUE TITLE
FlxColorUtil and FlxColor

### ISSUE THREAD

**Issue by JoeCreates** [HaxeFlixel/flixel#1027]
I'm wondering if FlxColor and FlxColorUtil are a little overcomplicated. FlxColor, which is likely the first place anyone would go to try to make a color, doesn't contain anything but presets. To make a color from rgb values you have to go to FlxColorUtil, at which point you are flooded with options, some of which seem redundant.

For example, let's say we just want to make a simple color from RGB values, and possibly an alpha. We can use:-
- FlxColorUtil.makeFromARGB(Alpha:Float, Red:Int, Green:Int, Blue:Int)
- FlxColorUtil.getColor32(Alpha:Int, Red:Int, Green:Int, Blue:Int)
- FlxColorUtil.getColor24(Red:Int, Green:Int, Blue:Int)

Why not something nice and simple in FlxColor? Like this perhaps:
- FlxColor.RGB(Red:Int, Green:Int, Blue:Int, Alpha:Int = 255)

Then we have things like:-
- FlxColorUtil.ARGBtoHexString(Color:Int):String
- FlxColorUtil.ARGBtoWebString(Color:Int):String
- FlxColorUtil.colorToHexString(Color:Int):String

Neither of the first two specifically take ARGB values, and yet they have ARGB in the name. The third isn't a "color" to hex at all, rather it is a byte (or value from 0-255) to hex string, and is used by the first two, but isn't really a color only thing.

Then there's a load of permutations for conversions of color representations:-
- ARGBtoRGB
- RGBtoHSV
- HSVtoARGB

Are these useful? ARGB and RGB are both just ints, and all this "conversion" does is set the alpha to 255. Seeing as there already exists methods for getting HSV or ARGB objects from color ints, the others also don't seem necessary.

If there were simply functions for getting colors for each method (getRGB, getHSV, etc..) that return an argb Int, and "to" functions for turning an Int into the desired representation, then every combination of conversions would be possible by doing things like toHSV(RGB(r, g, b)).

In summary, the colors stuff seems a bit excessive. I'm wondering if it might be possible to make it a bit for efficient. Perhaps, FlxColorUtil could in fact be merged into just FlxColor?
